File: python/src/dominio/compuesto.py
from src.dominio.componente import Componente


class Compuesto(Componente):
    __componentes = []

    def operacion(self):
        for componente in self.__componentes:
            componente.operacion()

    def agregar(self, componente):
        self.__componentes.append(componente)

    def eliminar(self, componente):
        self.__componentes.remove(componente)

File: java/src/main/java/dominio/Compuesto.java
package dominio;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Compuesto extends Componente {
    private List<Componente> componentes;

    public Compuesto(){
        this.componentes = new ArrayList<>();
    }

    @Override
    public void operacion(){
        this.componentes.forEach(c -> c.operacion());
    }

    public void agregar(Componente ... componentes){
        Collections.addAll(this.componentes, componentes);
    }

    public void eliminar(Componente componente){
        this.componentes.remove(componente);
    }
}